### PR TITLE
add specs for users and posts

### PR DESCRIPTION
**Pull request by abdozayan12**
- Implement spec test for request
- Implement spec test for views

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      get '/posts/index'
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET /show' do
+    it 'returns http success' do
+      get '/posts/show'
+      expect(response).to have_http_status(:success)
+    end
```
Comment (Ghiftee) [inline spec/requests/posts_spec.rb]: - [ ] Good job writing tests for the `index` and `show` methods of posts, however for comprehensive testing, aside from getting the right HTTP status, please include tests that ensure the correct views and response body as stated in the requirements below:
Screenshot 2023-10-03 at 11 51 30 (2)

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    it 'returns http success' do
+      get '/users/index'
+      expect(response).to have_http_status(:success)
+    end
+  end
```
Comment (Ghiftee) [inline spec/requests/users_spec.rb]: - [ ] Please add tests for the `show` method also as required
- [ ] Good job writing tests for the `index` methods of users, however for comprehensive testing, aside from getting the right HTTP status, please include tests that ensure the correct views and response body as stated in the requirements below:
Screenshot 2023-10-03 at 11 51 30 (2)

File: config/routes.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root 'users#index'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users/:user_id/posts/:id', to: 'posts#index', as: 'user_post'
+  get '/users/:user_id/posts', to: 'posts#show', as: 'user_posts'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users', to: 'users#index', as: 'users'
```
Comment (Ghiftee) [inline config/routes.rb:7]: - [ ] When I run the tests for post, I get this error below. Please ensure you are testing with the right routes
Screenshot 2023-10-03 at 12 04 21

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @abdozayan12  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ No linter errors
✔️ Tests are added for `users` and `posts`

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review 👇🏽 _

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mikethreels — APPROVED** [review]
Hi Abd, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the required changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**